src/pagination/index.tsx

```diff
@@ -51,22 +42,61 @@ interface PaginationCache {
 	initialPage?: number;
 	initialPageSize?: number;
 	currentPage: number;
+	nodeWidths: { prev: number; next: number; current: number; sibling: number };
 	pageSize: number;
 	pageSizes: number[];
-	pageSizesData: ListOption[];
+	pageSizesResource: { resource: () => Resource; data: ListOption[] };
+	theme?: Theme | ThemeWithVariant;
+}
+
+function createMemoryResourceWithData<S = any>(data: S[]) {
```

Comment:
This was used as a helper in the examples but realistically you should be creating a single resource outside of your render and then passing the data to it. You shouldn't be creating a whole new resource each time.

So here, just do

```ts
const template = {
				read: (_, put, get) => {
					let data: any[] = get();
					put(0, data);
					return { data, total: data.length };
				}
			};

const resource = createResource(template);
```

Then when you use it, pass it as
```
resource={{ resource: () => resource, data }}
```

The helper for the first part will be coming very soon via framework, today hopefully

Reply:
Got it, thanks!